### ISSUE TITLE
Saving Records Using ActiveRecord::Encryption Calls `encrypt` Twice -- Is this expected?

### ISSUE THREAD

**Issue by KyFaSt** [rails/rails#46195]
### Steps to reproduce
Hi, my org is migrating to use ActiveRecord::Encryption from an internally written encryption library. We set up some logging and noticed that we were seeing 2x as many encryptions as decryptions so I set up a[ new minimal rails app](https://github.com/KyFaSt/encrypted-columns-demo) with a custom encryptor to see if I could replicate the issue and I found that records are being encrypted twice during save. 

![Pasted Graphic 1](https://user-images.githubusercontent.com/4666485/193887698-f85f0f80-08ad-4552-a30c-2c49048dbf8c.png)

Is this expected? Steps to reproduce in console:
```irb
> a = Article.new(title: "Active Record Encryption and You")
> a.save
```
[Test I](https://github.com/KyFaSt/encrypted-columns-demo/blob/main/test/models/article_test.rb) wrote to reproduce this scenario:

```ruby
require "test_helper"

class ArticleTest < ActiveSupport::TestCase
  setup do
    logger = Logger.new(STDOUT)
    Article.any_instance.stubs(:logger).returns(logger)
    logger.stubs(:debug).returns(true)
    Rails.logger = logger
  end
  test "Encryptor#encrypt is called twice on save" do

    # expected to happen once when the record is initialized
    Rails.logger.expects(:info).with("encrypting My Title")
    # expected to happen once when the record is saved
    Rails.logger.expects(:info).with("encrypting My Title")
    # I was not expecting it a third time when the record is saved
    Rails.logger.expects(:info).with("encrypting My Title")

    Rails.logger.expects(:info).with("decrypting...").once
    a = Article.create(title: "My Title")
  end
end

```

### Expected behavior
I expected `Encryptor#encrypt` to be called once during `save`

### Actual behavior
It looks like `Encryptor#encrypt` is being called twice during `save`

### System configuration
**Rails version**: 7.0.4

**Ruby version**: 3.1.2

sample rails app that exhibits this behavior: https://github.com/KyFaSt/encrypted-columns-demo/

**Comment by eileencodes** [comment]
@jorgemanrubia can you take a look at this?

**Comment by jorgemanrubia** [comment]
@KyFaSt thanks for reporting this. I [prepared a fix here](https://github.com/rails/rails/pull/46226), based on [this PR](https://github.com/rails/rails/pull/44625) by @jonathanhefner that was recently merged.

**Comment by zzak** [comment]
This should be fixed by #46256
